packages/@cdktf/hcl2cdk/test/helpers/convert.ts

```diff
@@ -0,0 +1,287 @@
+  fqn: ProviderFqn;
+  type: ProviderType;
+  path: PathToCopy;
+};
+
+const cdktfBin = path.join(__dirname, "../../../../cdktf-cli/bin/cdktf");
+const cdktfDist = path.join(__dirname, "../../../../../dist");
+
+export const binding = {
+  aws: {
+    fqn: "hashicorp/aws@ ~>3.74.0",
+    type: ProviderType.provider,
+    path: "providers/aws",
+  },
+  docker: {
+    fqn: "kreuzwerker/docker@ ~>2.15.0",
+    type: ProviderType.provider,
+    path: "providers/docker",
+  },
+  null: {
+    fqn: "hashicorp/null@ ~>3.1.0",
+    type: ProviderType.provider,
+    path: "providers/null",
+  },
+  google: {
+    fqn: "hashicorp/google@ ~>3.87.0",
+    type: ProviderType.provider,
+    path: "providers/google",
+  },
+  azuread: {
+    fqn: "hashicorp/azuread@ ~>2.17.0",
+    type: ProviderType.provider,
+    path: "providers/azuread",
+  },
+  local: {
+    fqn: "hashicorp/local@ ~>2.1.0",
+    type: ProviderType.provider,
+    path: "providers/local",
+  },
+  auth0: {
+    fqn: "alexkappa/auth0@ ~>0.26.2",
+    type: ProviderType.provider,
+    path: "providers/auth0",
+  },
+  datadog: {
+    fqn: "DataDog/datadog@ ~>3.8.1",
+    type: ProviderType.provider,
+    path: "providers/datadog",
+  },
+  kubernetes: {
+    fqn: "hashicorp/kubernetes@ ~>2.11.0",
+    type: ProviderType.provider,
+    path: "providers/kubernetes",
+  },
+  awsVpc: {
+    fqn: "terraform-aws-modules/vpc/aws@ ~>3.11.5",
+    type: ProviderType.module,
+    path: "modules/terraform-aws-modules/aws",
+  },
+};
+
+type AbsolutePath = string;
+const providerBindingCache: Record<
+  ProviderFqn,
+  Promise<AbsolutePath> | undefined
+> = {};
+const providerSchemaCache: Record<ProviderFqn, SchemaPromise> = {};
+
+async function generateBindings(
+  binding: ProviderDefinition
+): Promise<AbsolutePath> {
+  const tempDir = fs.mkdtempSync(path.join(os.tmpdir(), "cdktf-provider-"));
+  await fs.writeFile(
+    path.resolve(tempDir, "cdktf.json"),
+    JSON.stringify({
+      language: "typescript",
+      app: "npx ts-node main.ts",
+      terraformProviders:
+        binding.type === ProviderType.provider ? [binding.fqn] : [],
+      terraformModules:
+        binding.type === ProviderType.module ? [binding.fqn] : [],
+    })
+  );
+  await execa(cdktfBin, ["get"], { cwd: tempDir });
+
+  return path.resolve(tempDir, ".gen", binding.path);
+}
+
+async function copyBindingsForProvider(
+  binding: ProviderDefinition,
+  targetDirectory: AbsolutePath
+) {
+  const absoluteBindingPathPromise = providerBindingCache[binding.fqn]
+    ? providerBindingCache[binding.fqn]
+    : generateBindings(binding);
+
+  providerBindingCache[binding.fqn] = absoluteBindingPathPromise;
+
+  const target = path.resolve(targetDirectory, ".gen", binding.path);
+  await fs.mkdirp(target);
+  const absolutePath = await absoluteBindingPathPromise;
+
+  await fs.copy(absolutePath!, target);
+}
+
+// Prepare for tests / warm up cache
+const baseProjectPromise = new Promise<string>(async (resolve) => {
```

Comment:
I guess one of the benefits of doing this would be to run this code only once even after importing it multiple times.

Reply:
Will this also run if I target a different case and jest invokes all testcases to find the one I target?
If yes, maybe lazily starting this might prevent this?

Reply:
Yeah I'm also not a big fan of that, but here it made sense to me :) If you like we can put it into a beforeAll step on each of the test files, but that felt more complicated than it needed to be.

> Will this also run if I target a different case and jest invokes all test cases to find the one I target?

If you target, e.g. the expressions tests which don't use a project and schemas you don't import this file, nothing gets run. For all other test cases we need the base project and also the schemas / bindings at a certain point so not much one can do to get around it. 

I will test if not warming up the cache improves the runtime :)

Reply:
Not warming up does not change the runtime :(